### PR TITLE
fix(upload):break upload if any error is thrown from blobber

### PR DESCRIPTION
**Pull request by cnlangzi**
## Fixes
- fixed: https://github.com/0chain/gosdk/issues/266
- fixed: https://github.com/0chain/zboxcli/issues/130

## Associated PRs (Link as appropriate):
- zboxcli: https://github.com/0chain/zboxcli/pull/193
- gosdk: https://github.com/0chain/gosdk/pull/417

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/break_upload_on_error  |
| gosdk  | staging  |
| zboxcli  | fix/break_upload_on_error  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/break_upload_on_error  |
| gosdk  | fix/break_upload_on_error  |
| zboxcli  | fix/break_upload_on_error  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/break_upload_on_error  |
| gosdk  | fix/break_upload_on_error  |
| zboxcli  | fix/break_upload_on_error  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: tests/cli_tests/zboxcli_sync_test.go [hunk 1]
```diff
@@ -19,728 +18,728 @@ import (
...
+	// 	require.Nil(t, err, "Error in listing the allocation files: ", strings.Join(output, "\n"))
+	// 	require.Len(t, output, 1)
+
+	// 	var files []climodel.AllocationFile
+	// 	err = json.Unmarshal([]byte(output[0]), &files)
+	// 	require.Nil(t, err, "Error deserializing JSON string `%s`: %v", strings.Join(output, "\n"), err)
+
+	// 	// This will traverse the tree and asserts the existent of the files
+	// 	assertFileExistenceRecursively(t, mockFolderStructure, files)
+	// })
```
Comment (Kishan-Dhakan) [inline tests/cli_tests/zboxcli_sync_test.go:742]: These need to be uncommented before merge

File: tests/cli_tests/zboxcli_sync_test.go [hunk 2]
```diff
@@ -770,13 +769,12 @@ func TestSyncWithBlobbers(t *testing.T) {
 		}, true)
 		require.Nil(t, err, "Error in syncing the folder: ", strings.Join(output, "\n"))
 
-		require.GreaterOrEqual(t, len(output), 4, "unexpected number of lines in output", strings.Join(output, "\n"))
-
-		require.Contains(t, output[len(output)-3], "Error in file operation: Upload failed:", output[len(output)-4])
-		require.Contains(t, output[len(output)-2], "Upload failed", output[len(output)-3])
+		fmt.Println(strings.Join(output, ""))
```
Comment (Kishan-Dhakan) [inline tests/cli_tests/zboxcli_sync_test.go:772]: Debug line needs to be removed

**Review by Kishan-Dhakan — CHANGES_REQUESTED** [review]
LGTM pending minor changes

**Review by Kishan-Dhakan — APPROVED** [review]

**Pull request merged**